File: src/main/java/fi/metatavu/metaform/server/persistence/dao/ListReplyFieldDAO.java
package fi.metatavu.metaform.server.persistence.dao;

import java.util.UUID;

import javax.enterprise.context.ApplicationScoped;

import fi.metatavu.metaform.server.persistence.model.ListReplyField;
import fi.metatavu.metaform.server.persistence.model.Reply;

/**
 * DAO class for ListReplyField entity
 * 
 * @author Antti Leppä
 */
@ApplicationScoped
public class ListReplyFieldDAO extends ReplyFieldDAO<ListReplyField> {
  
  /**
   * Creates new ListReplyField 
   * 
   * @param id id
   * @param reply reply
   * @param name field name
   * @param value field value
   * @return created field
   */
  public ListReplyField create(UUID id, Reply reply, String name) {
    ListReplyField replyField = new ListReplyField(); 
    replyField.setId(id);
    replyField.setName(name);
    replyField.setReply(reply);
    return persist(replyField);
  }
  
}

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/dao/ListReplyFieldDAO.kt
package fi.metatavu.metaform.server.persistence.dao

import fi.metatavu.metaform.server.persistence.model.ListReplyField
import fi.metatavu.metaform.server.persistence.model.Reply
import java.util.*
import jakarta.enterprise.context.ApplicationScoped

/**
 * DAO class for ListReplyField entity
 *
 * @author Antti Leppä
 */
@ApplicationScoped
class ListReplyFieldDAO : ReplyFieldDAO<ListReplyField>() {

  /**
   * Creates new ListReplyField
   *
   * @param id id
   * @param reply reply
   * @param name field name
   * @return created field
   */
  fun create(id: UUID, reply: Reply, name: String): ListReplyField {
    val replyField = ListReplyField()
    replyField.id = id
    replyField.name = name
    replyField.reply = reply
    return persist(replyField)
  }
}